packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1961,14 +1963,14 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
             observer.didRemove(removedRoute, newPrecedingRoute);
           removedRoute.dispose();
         }
-
-        if (newPrecedingRoute != null)
-          newPrecedingRoute.didChangeNext(newRoute);
       }
     });
 
     // Notify for newRoute
     newRoute.didChangeNext(null);
+    if (precedingRoute != null) {
```

Comment:
This is actually undoing the changes from #35223. IIRC, the new route needs to finish pushing before we can notify the preceding route, which is why it is inside of `whenCompleteOrCancel`.